### ISSUE TITLE
In-graph tracing: handle more nodes and graph types

### ISSUE THREAD

**Issue by jotak** [kiali/kiali#3048]
Epic: #2677 

Currently, the entry point to show traces is Service nodes.

First thing: it's actually wrong as pointed out there: https://github.com/kiali/kiali/issues/3045 . It just works accidentally when service name == app label. So the real entry point should be apps.

Second thing: we can expand this a little bit, by allowing any kind of entry point as long as there's an app label defined (it could work for apps, services and workloads nodes, although the last two need some post-fetch filtering)

**Comment by jshaughn** [comment]
@jotak As mentioned in my comment in #3045 the "app" is typically set for Service, App and Workload nodes when using telemetry V2.  This is because canonical_service is always set in the telemetry, defaulting to workload name.  Of course "app" may not be set on certain [service] nodes that don't directly tie to a workload.

**Comment by jotak** [comment]
Yes, and in jaeger it's similar to telemetry: when "app" is not set, then the workload name is used instead as the jaeger service identifier. So if we want to go further, when app label isn't defined, we could also get traces from workload names.

**Comment by jshaughn** [comment]
I like service nodes because they can appear in any sort of graph type.  But feasibly any node with app set, which would include workload nodes as well.
